### PR TITLE
Add Clang builds to Travis CI

### PR DESCRIPTION
**Pull request by changlan**
- Add Clang builds to Travis CI
- PEP-8 formatting

### CODE REVIEW

File: .travis/rebuild_image.sh [hunk 1]
```diff
@@ -4,7 +4,7 @@ set -e
 
 REPOSITORY=${REPOSITORY:-nefelinetworks/bess_build}
 VERSION=${VERSION:-`date +%y%m%d`}
-docker build --build-arg BESS_DPDK_BRANCH=${BESS_DPDK_BRANCH:-develop} -t $REPOSITORY:latest -t $REPOSITORY:$VERSION .
+docker build --build-arg BESS_DPDK_BRANCH=${BESS_DPDK_BRANCH:-c++} -t $REPOSITORY:latest -t $REPOSITORY:$VERSION .
```
Comment (justinemarie) [inline .travis/rebuild_image.sh]: Aren't we moving c++ to develop soon? Why c++ now?

File: build.py [hunk 2]
```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/env python2.7
+#!/usr/bin/env python
```
Comment (justinemarie) [inline build.py]: Is this recommended? I know that between versions of python there are often compatibility issues...

**Comment by justinemarie** [comment]
Most of these changes are due to formatting script...

File: .travis/rebuild_image.sh (hunk 1 above)
Comment (changlan) [inline .travis/rebuild_image.sh]: This script broke due to a previous change to the develop branch. We should keep it working.
Also: perhaps we won't be renaming c++ to develop.

File: build.py (hunk 2 above)
Comment (changlan) [inline build.py]: - In general I think we should support Python 2.x (which we do). I don't think we used anything 2.7-specific so it'd be better to keep it general.

- python2.7 is a Linux distro-specific name.

**Comment by changlan** [comment]
Yup, those PEP-8 formatting changes are intentional. :)

**Comment by melvinw** [comment]
LGTM

**Comment by melvinw** [comment]
cat
Thanks!

**Pull request merged**